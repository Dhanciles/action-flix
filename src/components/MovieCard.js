import React from 'react'

export const MovieCard = ({title, rating, releaseDate, description}) => {
  return (
    <article>
      <p>{rating}</p>
      <h3>{title}</h3>
      <p>{releaseDate}</p>
      <p>{description}</p>
    </article>
  )
}