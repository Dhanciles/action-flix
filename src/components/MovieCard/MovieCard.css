.movie-card {
  border: 5px solid #8E959C;
  display: flex;
  height: 300px;
  margin-bottom: 25px;
  padding-right: 10px;
  width: 500px;
}

.movie-img {
  height: 300px;
  margin-right: 20px;
  width: 185px;
}

.movie-info-text {
  width: 100%;
}

.movie-icon-container {
  display: flex;
  justify-content: space-between;
  padding: 0 0 5px 0;
}

.movie-rating-chart {
  height: 50px;
  padding-top: 12px;
  width: 50px;
}

.outline-circle,
.circle {
  stroke-width: 3;
}

.outline-circle {
  fill: #8E959C;
  stroke: #fff;
}

.circle {
  fill: none;
  animation: progress 4s ease-out forwards;
  stroke: #76E1EF;
  stroke-linecap: round;
}

@keyframes progress {
  0% {
    stroke-dasharray: 0 100;
  }

  60% {
    stroke-dasharray: 100 100;
  }
}

.movie-rating {
  font-size: 0.8rem;
}

.popcorn-icon {
  height: 60px;
  /*margin-bottom: 10px;*/
}

.movie-title, 
.movie-release {
  padding-left: 5px;
}

.movie-title {
  color: #76E1EF;
  font-size: 1.5rem;
  margin: 0;
}

.movie-release,
.movie-description {
  color: #8E959C;
}

.movie-description {
  border-radius: 10px;
  cursor: default;
  overflow: auto;
  width: 200px;
  height: 100px;
  transition: box-shadow 500ms;
  padding: 5px;
}

.movie-description:hover {
  box-shadow: inset 0 -45px 130px -55px rgba(0, 0, 0, 1);
}

