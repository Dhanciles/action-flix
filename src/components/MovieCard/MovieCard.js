import React from 'react'
import './MovieCard.css'
import { connect } from 'react-redux'


export const MovieCard = ({title, posterPath, releaseDate, voteAverage, overview, props}) => {
  return (
    <article className='movie-card'>
      <img className='movie-img' src={posterPath} alt='movie poster' />
      <div className='movie-info-text'>
        <div className='movie-icon-container'>
          <p className='movie-rating'>{voteAverage}</p>
          <img src='../images/full-popcorn.svg' alt='full popcorn' onClick={() => props.addFavorite(movieId, props.userId, title, posterPath, releaseDate, voteAverage, overview)} />
        </div>
        <h3 className='movie-title'>{title}</h3>
        <p className='movie-release'>{releaseDate}</p>
        <p className='movie-description'>{overview}</p>
      </div>
    </article>
  )

  export const mapDispatchToProps = (dispatch) => ({
    addFavorite: () => dispatch(postFavorite(movieId, userId, title, posterPath, releaseDate, voteAverage, overview))
  })

  export const mapStateToProps = (state) => ({
    userId: state.userId
  })

  export default connect(mapStateToProps, mapDispatchToProps)(MovieCard)
}