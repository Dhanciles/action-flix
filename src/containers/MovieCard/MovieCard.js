import React from 'react'
import './MovieCard.css'
import { Component } from 'react'
import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import { postFavorite } from '../../thunks/postFavorite.js'
import { deleteFavorite } from '../../thunks/deleteFavorite.js'
import { toggleFavorite } from '../../actions/index.js'
import PropTypes from 'prop-types'



export class MovieCard extends Component {

  checkSignedIn = () => {
    if (this.props.userId === '') {
      this.props.history.push(`/login`)
    } else {
      const isFavorite = this.props.movies.find(movie => {
        return movie.movieId === this.props.movieId
      })
      if(!isFavorite.favorite) {
        this.props.addFavorite(this.props.movieId, this.props.userId, this.props.title, this.props.posterPath, this.props.releaseDate, this.props.voteAverage, this.props.overview)
        this.props.toggleFavorite(this.props.title)
      } else {
        this.props.toggleFavorite(this.props.title)
        this.props.deleteFavorite(this.props.userId, this.props.movieId)
      }
    }
  }

  render() {
   const favoriteIcon = this.props.favorite ? '../images/full-popcorn.svg' : '../images/empty-popcorn.svg'
    return (
      <article className='movie-card'>
        <img className='movie-img' src={this.props.posterPath} alt='movie poster' />
        <div className='movie-info-text'>
          <div className='movie-icon-container'>
            <svg className='movie-rating-chart' viewBox="0 0 36 36">
              <path className='outline-circle'
              d="M18 2.0845
              a 15.9155 15.9155 0 0 1 0 31.831
              a 15.9155 15.9155 0 0 1 0 -31.831"
              strokeDasharray="100, 100"               
              />
              <path className='circle'
              d="M18 2.0845
              a 15.9155 15.9155 0 0 1 0 31.831
              a 15.9155 15.9155 0 0 1 0 -31.831"
              strokeDasharray={`${this.props.voteAverage * 10}, 100`}               
              />
              <text textAnchor="middle" x="17" y="22" className='movie-rating'>{this.props.voteAverage}</text>
            </svg>
            <img className='popcorn-icon' src={favoriteIcon} alt='full popcorn' onClick={() => this.checkSignedIn()} />
          </div>
          <h3 className='movie-title'>{this.props.title}</h3>
          <p className='movie-release'>{this.props.releaseDate}</p>
          <div className='movie-description'>
            {this.props.overview}
          </div>
        </div>
      </article>
    )
  }
}

MovieCard.propTypes = {
  addFavorite: PropTypes.func.isRequired, 
  toggleFavorite: PropTypes.func.isRequired, 
  deleteFavorite: PropTypes.func.isRequired,
  movies: PropTypes.array.isRequired, 
  userId: PropTypes.number
}


export const mapDispatchToProps = (dispatch) => ({
  addFavorite: (movieId, userId, title, posterPath, releaseDate, voteAverage, overview) => dispatch(postFavorite(movieId, userId, title, posterPath, releaseDate, voteAverage, overview)),
  toggleFavorite: (movieTitle) => dispatch(toggleFavorite(movieTitle)),
  deleteFavorite: (userId, movieId) => dispatch(deleteFavorite(userId, movieId))
})

export const mapStateToProps = (state, props) => ({
  movies: state.movies,
  userId: state.userId,
  movieId: props.movieId, 
  title: props.title, 
  posterPath: props.posterPath, 
  releaseDate: props.releaseDate, 
  voteAverage: props.voteAverage, 
  overview: props.overview, 
  favorite: props.favorite
})

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(MovieCard))

