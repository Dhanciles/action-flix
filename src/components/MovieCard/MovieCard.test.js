import React from 'react'
import { shallow } from 'enzyme'
import { MovieCard, mapStateToProps, mapDispatchToProps } from './MovieCard'

describe('MovieCard', () => {
  it('should match the snapshot', () => {
    let mockMovies = [{title: 'Aquaman'}, {title: 'Transformers'}]
    let mockUser = 1
    const wrapper = shallow(<MovieCard movies={mockMovies} userId={mockUser} />)

    expect(wrapper).toMatchSnapshot();
  });

  describe('checkSignedIn', () => {
    let wrapper
    let mockMovies
    let mockUser
    let mockMovieId
    let mockAddFavorite
    let mockDeleteFavorite
    let mockToggleFavorite


    beforeEach(() => {
      mockUser = 1
      mockMovieId = 1
      mockAddFavorite = jest.fn()
      mockDeleteFavorite = jest.fn()
      mockToggleFavorite = jest.fn()
      mockMovies = [{
        title: 'Aquaman', 
        movieId: 1, 
        posterPath: 'www.thismovie.com', 
        releaseDate: '12/12/18',
        voteAverage: 7.5,
        overview: 'About this movie',
        favorite: false
      }, 
      {
        title: 'Transformers', 
        movieId: 2, 
        posterPath: 'www.thismovie.com', 
        releaseDate: '12/12/18',
        voteAverage: 7.5,
        overview: 'About this movie',
        favorite: false
      }]
      wrapper = shallow(<MovieCard movies={mockMovies} userId={mockUser} movieId={mockMovieId} addFavorite={mockAddFavorite} deleteFavorite={mockDeleteFavorite} toggleFavorite={mockToggleFavorite} />)
    })

    it('should invoke checkSignedIn when the favorite', () => {
      wrapper.checkSignedIn = jest.fn()
      wrapper.find('.popcorn').simulate('click')

      expect(wrapper.checkSignedIn).toHaveBeenCalled
    })

    it('should call addFavorite if the movie is not a favorite', () => {
      wrapper.checkSignedIn = jest.fn()
      wrapper.instance().checkSignedIn()

      expect(wrapper.addFavorite).toHaveBeenCalled
    })

    it('should call toggleFavorite', () => {
      wrapper.checkSignedIn = jest.fn()
      wrapper.instance().checkSignedIn()

      expect(wrapper.toggleFavorite).toHaveBeenCalled
    })

    it('should call deleteFavorite', () => {
      wrapper.checkSignedIn = jest.fn()
      wrapper.instance().checkSignedIn()

      expect(wrapper.deleteFavorite).toHaveBeenCalled
    })
  })
})