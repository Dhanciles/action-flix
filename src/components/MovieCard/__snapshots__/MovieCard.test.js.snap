// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="movie poster"
          className="movie-img"
        />,
        <div
          className="movie-info-text"
        >
          <div
            className="movie-icon-container"
          >
            <p
              className="movie-rating"
            />
            <img
              alt="full popcorn"
              className="favorite-icon"
              onClick={[Function]}
              src="../images/empty-popcorn.svg"
            />
          </div>
          <h3
            className="movie-title"
          />
          <p
            className="movie-release"
          />
          <p
            className="movie-description"
          />
        </div>,
      ],
      "className": "movie-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie poster",
          "className": "movie-img",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="movie-icon-container"
            >
              <p
                className="movie-rating"
              />
              <img
                alt="full popcorn"
                className="favorite-icon"
                onClick={[Function]}
                src="../images/empty-popcorn.svg"
              />
            </div>,
            <h3
              className="movie-title"
            />,
            <p
              className="movie-release"
            />,
            <p
              className="movie-description"
            />,
          ],
          "className": "movie-info-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="movie-rating"
                />,
                <img
                  alt="full popcorn"
                  className="favorite-icon"
                  onClick={[Function]}
                  src="../images/empty-popcorn.svg"
                />,
              ],
              "className": "movie-icon-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "movie-rating",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "full popcorn",
                  "className": "favorite-icon",
                  "onClick": [Function],
                  "src": "../images/empty-popcorn.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "movie-title",
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "movie-release",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "movie-description",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="movie poster"
            className="movie-img"
          />,
          <div
            className="movie-info-text"
          >
            <div
              className="movie-icon-container"
            >
              <p
                className="movie-rating"
              />
              <img
                alt="full popcorn"
                className="favorite-icon"
                onClick={[Function]}
                src="../images/empty-popcorn.svg"
              />
            </div>
            <h3
              className="movie-title"
            />
            <p
              className="movie-release"
            />
            <p
              className="movie-description"
            />
          </div>,
        ],
        "className": "movie-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "className": "movie-img",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="movie-icon-container"
              >
                <p
                  className="movie-rating"
                />
                <img
                  alt="full popcorn"
                  className="favorite-icon"
                  onClick={[Function]}
                  src="../images/empty-popcorn.svg"
                />
              </div>,
              <h3
                className="movie-title"
              />,
              <p
                className="movie-release"
              />,
              <p
                className="movie-description"
              />,
            ],
            "className": "movie-info-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="movie-rating"
                  />,
                  <img
                    alt="full popcorn"
                    className="favorite-icon"
                    onClick={[Function]}
                    src="../images/empty-popcorn.svg"
                  />,
                ],
                "className": "movie-icon-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "movie-rating",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "full popcorn",
                    "className": "favorite-icon",
                    "onClick": [Function],
                    "src": "../images/empty-popcorn.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "movie-title",
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "movie-release",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "movie-description",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
